File: 3747-maximum-difference-between-adjacent-elements-in-a-circular-array/maximum-difference-between-adjacent-elements-in-a-circular-array.rb
# @param {Integer[]} nums
# @return {Integer}
def max_adjacent_distance(nums)
    result = (nums[0] - nums[-1]).abs
    (0...nums.length - 1).each do |i|
        result = [(nums[i] - nums[i + 1]).abs, result].max
    end
    result
end

File: 3747-maximum-difference-between-adjacent-elements-in-a-circular-array/maximum-difference-between-adjacent-elements-in-a-circular-array.go
func maxAdjacentDistance(nums []int) int {
    result := abs(nums[0] - nums[len(nums) - 1]);
    for i := 0; i < len(nums) - 1; i++ {
        result = max(result, abs(nums[i] - nums[i + 1]));
    }

    return result;
}

func abs(a int) int {
    if a < 0 {
        return -a;
    }
    return a;
}